src/sentry/static/sentry/app/components/search/index.jsx

```diff
@@ -0,0 +1,192 @@
+     *  index: item's index in results
+     *  highlighted: is item highlighted
+     *  itemProps: props that should be spread for root item
+     * }
+     */
+    renderItem: PropTypes.func,
+    dropdownStyle: PropTypes.string,
+    searchOptions: PropTypes.object,
+  };
+
+  static defaultProps = {
+    // Default Search result rendering
+    renderItem: ({item, matches, itemProps, highlighted}) => (
+      <SearchResultWrapper {...itemProps} highlighted={highlighted}>
+        <SearchResult item={item} matches={matches} />
+      </SearchResultWrapper>
+    ),
+  };
+
+  handleSelect = (item, state) => {
+    if (!item) return;
+
+    let {to} = item;
+    if (!to) return;
+
+    let {params, router} = this.props;
+    let nextPath = replaceRouterParams(to, params);
+
+    navigateTo(nextPath, router);
+  };
+
+  renderItem = ({resultObj, index, highlightedIndex, getItemProps}) => {
+    // resultObj is a fuse.js result object with {item, matches, score}
+    let {renderItem} = this.props;
+    let highlighted = index === highlightedIndex;
+    let {item, matches} = resultObj;
+    let key = `${item.title}-${index}`;
+    let itemProps = {
+      ...getItemProps({
+        item,
+      }),
+    };
+
+    if (typeof renderItem !== 'function') {
+      throw new Error('Invalid `renderItem`');
+    }
+
+    return React.cloneElement(
+      renderItem({
+        item,
+        matches,
+        index,
+        highlighted,
+      }),
+      {
+        ...itemProps,
+        key,
+      }
+    );
+  };
+
+  render() {
+    let {
+      params,
+      dropdownStyle,
+      searchOptions,
+      minSearch,
+      maxResults,
+      renderInput,
+    } = this.props;
+
+    return (
+      <AutoComplete
+        defaultHighlightedIndex={0}
+        itemToString={() => ''}
+        onSelect={this.handleSelect}
+      >
+        {({
+          getInputProps,
+          getItemProps,
+          isOpen,
+          inputValue,
+          selectedItem,
+          highlightedIndex,
+          onChange,
+        }) => {
+          let searchQuery = inputValue.toLowerCase();
+          let isValidSearch = inputValue.length >= minSearch;
+
+          return (
+            <SearchWrapper>
+              {renderInput({
+                getInputProps,
+              })}
+
+              {isValidSearch && isOpen ? (
+                <SearchSources
+                  searchOptions={searchOptions}
+                  query={searchQuery}
+                  params={params}
+                >
+                  {({isLoading, results, hasAnyResults}) => (
+                    <DropdownBox css={dropdownStyle}>
+                      {isLoading && (
+                        <Flex justify="center" align="center" p={1}>
+                          <LoadingIndicator mini hideMessage relative />
+                        </Flex>
+                      )}
+                      {!isLoading &&
+                        results.splice(0, maxResults).map((resultObj, index) => {
```

Comment:
wait, is this supposed to be slice not splice?

Reply:
how did that get there